src/ip/Intercept.cc

```diff
@@ -157,22 +157,20 @@ Ip::Intercept::NetfilterInterception(const Comm::ConnectionPointer &newConn)
     return false;
 }
 
+/// whether TPROXY was enabled by configure
 bool
-Ip::Intercept::TproxyTransparent(const Comm::ConnectionPointer &newConn)
+Ip::Intercept::tproxyEnabled() const
 {
+    /* --enable-linux-netfilter    */
+    /* --enable-pf-transparent     */
+    /* --enable-ipfw-transparent   */
 #if (LINUX_NETFILTER && defined(IP_TRANSPARENT)) || \
     (PF_TRANSPARENT && defined(SO_BINDANY)) || \
     (IPFW_TRANSPARENT && defined(IP_BINDANY))
-
-    /* Trust the user configured properly. If not no harm done.
-     * We will simply attempt a bind outgoing on our own IP.
-     */
-    debugs(89, 5, "address TPROXY: " << newConn);
     return true;
-#else
-    (void)newConn;
-    return false;
 #endif
+    debugs(89, DBG_IMPORTANT, "WARNING: transparent proxying not supported");
```

Comment:
Feels odd to warn about X being unsupported in a function that tests whether X is supported. Let the caller decide whether the lack of support is important enough to warn about.


```suggestion
```